### PR TITLE
[MO/JS] - [it tests] -> smoke tests done

### PR DESCRIPTION
**Pull request by Seequick1**
### Type of change 
* feature

### Description
This PR also add some basic tests for `confluent` and `apicurio` artifacts, metadata, versions and rules.

### CODE REVIEW

**Comment by Frawless** [comment]
@alesj @EricWittmann could you guys review this? We want to merge it ASAP to be able continue on two different types of tests (we need base merged before it). Startup of Kafka and Registry before each test will be changed in the future PR based on the test scenario. Also we will improve option to run tests against specific registry app (running in OCP for example)

**Comment by EricWittmann** [comment]
I ran the tests locally and they all passed.  So that's good. :)  After merging we should add these to the CircleCI build.

File: tests/src/main/java/io/apicurio/tests/RegistryFacade.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    public static final String DEFAULT_REGISTRY_URL = "localhost";
+
+    public static final String REGISTRY_JAR_PATH = System.getenv().getOrDefault("REGISTRY_JAR_PATH", DEFAULT_REGISTRY_JAR_PATH);
+    public static final String REGISTRY_PORT = System.getenv().getOrDefault("REGISTRY_PORT", DEFAULT_REGISTRY_PORT);
+    public static final String REGISTRY_URL = System.getenv().getOrDefault("REGISTRY_URL", DEFAULT_REGISTRY_URL);
+    public static final String EXTERNAL_REGISTRY = System.getenv().getOrDefault("EXTERNAL_REGISTRY", "");
+
+    private Exec executor = new Exec();
+
+    private Future<Boolean> registries;
```
Comment (alesj) [inline tests/src/main/java/io/apicurio/tests/RegistryFacade.java]: Registries, as in many of them? There should be only one ...
With different Storage type -- in-memory (as default), JPA, Kafka, Streams, Infinispan, ...

File: tests/src/main/java/io/apicurio/tests/utils/subUtils/ArtifactUtils.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+import io.apicurio.tests.utils.BaseHttpUtils;
+import io.restassured.path.json.JsonPath;
+import io.restassured.response.Response;
+
+import java.util.HashMap;
+
+import static io.apicurio.tests.utils.BaseHttpUtils.getRequest;
+import static io.apicurio.tests.utils.BaseHttpUtils.putRequest;
+
+public class ArtifactUtils {
```
Comment (alesj) [inline tests/src/main/java/io/apicurio/tests/utils/subUtils/ArtifactUtils.java]: Use the client ...

File: tests/src/main/java/io/apicurio/tests/utils/subUtils/TestUtils.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                if (writer != null) {
+                    writer.close();
+                }
+            } catch (Exception ex) {
+                ex.printStackTrace();
+            }
+        }
+    }
+
+    public static String ruleToString(Rule rule) {
```
Comment (alesj) [inline tests/src/main/java/io/apicurio/tests/utils/subUtils/TestUtils.java]: I'm sure there is some JSON lib on the classpath, so you don't need to do this manually.

File: tests/src/test/java/io/apicurio/tests/BaseIT.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TimeZone;
+
+public abstract class BaseIT implements TestSeparator, Constants {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BaseIT.class);
+    private static KafkaFacade kafkaCluster = new KafkaFacade();
```
Comment (alesj) [inline tests/src/test/java/io/apicurio/tests/BaseIT.java]: Not all storage types need Kafka ...

**Pull request merged**